kvdb/src/lib.rs

```diff
@@ -70,48 +62,37 @@ impl DBOp {
 	}
 }
 
-impl DBTransaction {
+impl<K, V> DBTransaction<K, V> {
 	/// Create new transaction.
-	pub fn new() -> DBTransaction {
+	pub fn new() -> Self {
 		DBTransaction::with_capacity(256)
 	}
 
 	/// Create new transaction with capacity.
-	pub fn with_capacity(cap: usize) -> DBTransaction {
+	pub fn with_capacity(cap: usize) -> Self {
 		DBTransaction {
 			ops: Vec::with_capacity(cap)
 		}
 	}
 
 	/// Insert a key-value pair in the transaction. Any existing value will be overwritten upon write.
-	pub fn put(&mut self, col: Option<u32>, key: &[u8], value: &[u8]) {
-		let mut ekey = ElasticArray32::new();
-		ekey.append_slice(key);
+	pub fn put<IK, IV>(&mut self, col: Option<u32>, key: IK, value: IV)
```

Comment:
Is defining `IK` and `IV` with into conversions useful here?

Reply:
I believe it is better than without.

Just to be sure, if you mean:
* we should directly have `K` and `V`, then this is for compatibility reason (less code change when migrating to this lib, `&[u8]` will probably work for most cases).
* we should keep only `&[u8]` everywhere, then the `Into` allows for more efficient code (one can transfer ownership of some `K` or `V` directly?)

Note that there may be even better ways.